### PR TITLE
Enhanced UI for fields and column of permissions and roles

### PR DESCRIPTION
**Pull request by shaoshiva**
Hi,

This pull request implements the fields and columns provided by https://github.com/Laravel-Backpack/CRUD/pull/943. See the latter for explanation on how it works.

### CODE REVIEW

**Comment by shaoshiva** [comment]
I just noticed that this PR implements the `prefix` and `item` methods on the `Permission` model, which are used in the views of the other PR, and the other PR requires the views implemented by this one. Maybe I should implement these methods in a separate PR that could be merged before, so the other PR will no longer be dependent on this one.

File: src/app/Models/Permission.php [hunk 1]
```diff
@@ -10,4 +10,30 @@ class Permission extends OriginalPermission
     use CrudTrait;
 
     protected $fillable = ['name', 'updated_at', 'created_at'];
+
+    /**
+     * Gets the permission prefix (eg. admin.page)
+     *
+     * @return null
```
Comment (lloy0076) [inline src/app/Models/Permission.php]: No it doesn't!

**Comment by tabacitu** [comment]
Hi @shaoshiva - this is some great work. 

As I mentioned [here](https://github.com/Laravel-Backpack/CRUD/pull/943#issuecomment-351291875), I think this is beyond the scope of PermissionManager. But I'd love to see this as a separate, more advanced Backpack package, something like AdvancedPermissionManager, if you want to create and maintain it.

Since you've already done most of the work, I'm sure you can find some more Backpack developers on Gitter to help maintain it (me included, of course).

Cheers!

**Comment by tabacitu** [comment]
PS. If I'm not wrong, I think @OwenMelbz also likes this approach - create/read/update/delete permissions for each entity. Your screenshot here https://github.com/Laravel-Backpack/CRUD/pull/943 might interest him :-)

**Pull request closed without merging**